### PR TITLE
[FIX] User status update 

### LINKED ISSUE #3003: User status should update only after save action
### Description:
User status (online, offline, e.t.c) will update even if a user closes the screen(If there is any change in status) without save, whereas user status text remains the same

### Environment Information:

- Rocket.Chat Server Version: 3.12
- Rocket.Chat App Version: 4.15
- Device Name: IPhone 11
both in Android & IOS

### Steps to reproduce:

1.  Click on your status (or edit status) from the sidebar.
2.  Change status (Online, offline)
3.  Change status text
4. Now hit "Cancel"
5. Observe your status(online, offline) will be changed, but user status text remains the same.  

### Expected behavior:
User status should not update if user press "Cancel"

### Actual behavior:
User status will update even if user press "cancel"

https://user-images.githubusercontent.com/23723464/111987463-44e02580-8b35-11eb-9cbf-be9f306e06ca.mov

### PR DESCRIPTION
**Pull request by gerzonc**
## Proposed changes

Based on #3004
Currently we're updating the user status two times: when we tap on the user status and when we tap on `Done`. This fixes the first one, since it updates the user status even if you tap on `Cancel`
## Issue(s)	

Closes #3003
## How to test or reproduce

Test case 1

1. Go to Sidebar > Edit user status
2. Change user status / user status text
3. Tap on `Cancel`

Expected behavior: User status / user status text should be the same.

Test case 2

1. Go to Sidebar > Edit user status
2. Change user status / user status text
3. Tap on `Done`

Expected behavior: User status / user status text should be updated.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/StatusView.tsx [hunk 1]
```diff
@@ -88,10 +89,10 @@ class StatusView extends React.Component<IStatusViewProps, IStatusViewState> {
 
 	submit = async () => {
 		logEvent(events.STATUS_DONE);
-		const { statusText } = this.state;
+		const { statusText, status } = this.state;
 		const { user } = this.props;
-		if (statusText !== user.statusText) {
-			await this.setCustomStatus(statusText);
+		if (statusText !== user.statusText || status !== user.status) {
+			await this.setCustomStatus();
```
Comment (diegolmello) [inline app/views/StatusView.tsx:95]: This is a refactor, so we should merge `submit` and `setCustomStatus`

File: app/views/StatusView.tsx [hunk 2]
```diff
@@ -101,31 +102,51 @@ class StatusView extends React.Component<IStatusViewProps, IStatusViewState> {
...
-		const { user, dispatch } = this.props;
+	setCustomStatus = async () => {
+		const { statusText, status } = this.state;
+		const { dispatch, user } = this.props;
 
 		this.setState({ loading: true });
 
 		try {
-			const result = await RocketChat.setUserStatus(user.status, statusText);
+			const result = await RocketChat.setUserStatus(status as UserStatus, statusText as string);
```
Comment (diegolmello) [inline app/views/StatusView.tsx:112]: We probably don't need to cast to string there.

File: app/views/StatusView.tsx [hunk 3]
```diff
@@ -101,31 +102,51 @@ class StatusView extends React.Component<IStatusViewProps, IStatusViewState> {
...
 			if (result.success) {
-				logEvent(events.STATUS_CUSTOM);
-				dispatch(setUser({ statusText }));
+				if (statusText !== user.statusText) {
+					logEvent(events.STATUS_CUSTOM);
+				}
+				if (status !== user.status) {
+					// @ts-ignore
+					logEvent(events[`STATUS_${status.toUpperCase()}`]);
+				}
```
Comment (diegolmello) [inline app/views/StatusView.tsx:120]: Log user events.
We're looking for submit, tapped `away`, etc.

File: app/views/StatusView.tsx [hunk 4]
```diff
@@ -101,31 +102,51 @@ class StatusView extends React.Component<IStatusViewProps, IStatusViewState> {
...
-		} catch {
-			logEvent(events.STATUS_CUSTOM_F);
+		} catch (e: any) {
+			showErrorAlert(I18n.t(e.data.errorType));
+			if (statusText !== user.statusText) {
+				logEvent(events.STATUS_CUSTOM_F);
+			}
+			if (status !== user.status) {
+				logEvent(events.SET_STATUS_FAIL);
+			}
```
Comment (diegolmello) [inline app/views/StatusView.tsx:140]: No need to do this.
`log` is going to persist the relevant info.

**Review by diegolmello — CHANGES_REQUESTED** [review]

**Review by diegolmello — CHANGES_REQUESTED** [review]
Missing update on Detox tests

**Review by reinaldonetof — APPROVED** [review]
LGTM

**Pull request merged**